src/main/kotlin/org/cascadebot/cascadebot/commands/moderation/UnMuteCommand.kt

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2020 CascadeBot. All rights reserved.
+ * Licensed under the MIT license.
+ */
+
+package org.cascadebot.cascadebot.commands.moderation
+
+import net.dv8tion.jda.api.Permission
+import net.dv8tion.jda.api.entities.Member
+import org.cascadebot.cascadebot.commandmeta.CommandContext
+import org.cascadebot.cascadebot.commandmeta.MainCommand
+import org.cascadebot.cascadebot.commandmeta.Module
+import org.cascadebot.cascadebot.data.managers.ScheduledActionManager
+import org.cascadebot.cascadebot.messaging.MessagingObjects
+import org.cascadebot.cascadebot.permissions.CascadePermission
+import org.cascadebot.cascadebot.scheduler.ActionType
+import org.cascadebot.cascadebot.scheduler.ScheduledAction
+import org.cascadebot.cascadebot.utils.DiscordUtils
+import org.cascadebot.cascadebot.utils.getMutedRole
+
+class UnMuteCommand : MainCommand() {
+
+    override fun onCommand(sender: Member, context: CommandContext) {
+        if (context.args.isEmpty()) {
+            context.uiMessaging.replyUsage()
+            return
+        }
+
+        val targetMember = DiscordUtils.getMember(context.guild, context.getArg(0))
+
+        if (targetMember == null) {
+            context.typedMessaging.replyDanger(MessagingObjects.getStandardMessageEmbed(context.i18n("responses.cannot_find_user"), context.user))
+            return
+        }
+
+        val mutedRole = context.guild.getMutedRole()
+
+        if (targetMember.roles.contains(mutedRole)) {
+            ScheduledActionManager.removeIf {
+                if (it.type != ActionType.UNMUTE) return@removeIf false
+                if (it.data !is ScheduledAction.ModerationActionData) return@removeIf false
+                it.data.targetId == targetMember.idLong
+            }
+        } else {
```

Comment:
This seems as if we intend to unmute the member, but then we also don't reply when we have done it
We also don't include the reason to the audit log
Also we don't catch the error if we don't have permission to unmute the user

When adding reason to audit log, we just don't. It adds "{author} muted user", doesn't add reason. (This is across all of mute commands)

Reply:
We can't send a reason to the audit logs. It's a role add, they don't have a reason attached to them (well they kind of do, but not in a way that matters).
Also I'm fairly sure the moderation code handles the replying. Could be wrong, I can't look right now as I'm on mobile.

Reply:
Ryan is right in that we can't actually log to audit log (But I will still add a reason in case this is added on Discord's end)

Ikiinon is right that this doesn't actually mute 😅 I obviously got too excited over my removeIf statement 😆

Reply:
Don't understand what you mean in that you can't send a reason to audit logs in adding roles. If I'm understanding correctly you can, and in fact we already do this when we mute/unmute in temp mute, as so: https://i.imgur.com/pN9kZF3.png
My statement was about that we don't append the reason into this. I'd expect it to be {author} muted user: {reason}
If we are not adding the reason to the log, then other than maybe modlog, why have a reason parameter?

Reply:
Mod log. Also I had no idea discord actually displayed that. I thought it just eat it. I'll look more into this when I get home.